### ISSUE TITLE
[feature request] SSIM-based cost function as part of the standard set of loss functions

### ISSUE THREAD

**Issue by plopd** [pytorch/pytorch#9250]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description

SSIM-based cost function as part of the standard set of loss functions.
See example here: https://github.com/Po-Hsun-Su/pytorch-ssim

**Comment by SsnL** [comment]
as a user: this is kinda useful to me, so you have my vote

as a dev: we need enough usage before deciding that this should be in core. whats your thoughts on this, @soumith ?

**Comment by ezyang** [comment]
IMO, if one core dev thinks it's a good idea, in it should go :)

**Comment by SsnL** [comment]
Just want to clarify: I didn’t mean that I think this should be in core. Just that from me as a user’s perspective, this will save me some trouble because I once needed it and wouldn’t have to implement it myself if it were in core. A lot of things are like this (depending whether I happen to use them in my research) but not necesarily that we should put them in core.

**Comment by vadimkantorov** [comment]
I think PyTorch needs a `contrib` repo strategy for less stable stuff :) Hopefully, something comes along after 1.0? :)

**Comment by soumith** [comment]
@apaszke and I were discussing this and https://github.com/pytorch/pytorch/pull/9177

We definitely need a `torchcontrib` package that publishes itself every two weeks or so, and can include loss functions, layers, etc from the latest and greatest papers.

**Comment by vadimkantorov** [comment]
Some ideas for it:
 - opt-in voting function (so that importing or using a module would send a vote to the server, and by resampling votes you may see if a module is really popular) - so that the most popular modules may be considered for promotion to core
 - some suggested structure such as "one module - one file" (a bit similar to MATLAB File Central) and relaxed testing requirements
 - some reasonable way to declare per-module dependencies on other libraries, so that installation of the whole package doesn't turn into an unpredictable nightmare
 - some pre-agreed deprecation schedule: if an example started to break and no corrections are done within two months, module is declared deprecated (and is moved out of the main contrib package in one year?)
